Country selection

MattiaSilvestri opened the pull request:
Changes:
- user can choose between automatic geolocation and manual country selection;
- minimal cli;
- help message;
- flag for getting country list in alphabetical order.

FrnRmn commented on README.md:
```diff
@@ -4,3 +4,10 @@ Simple tool to monitor your personal computer power consumption and emissions.
 ### Useful Links
 - [Electricity Maps](https://app.electricitymaps.com/map)
 - [Co2Signal API](https://www.co2signal.com/)
+
```
next time I would do this changes outside the branch, just for future commits

MattiaSilvestri replied:
Not sure I got it, you mean to not update the README directly in dev or main?

FrnRmn replied:
No no, just that the README to-do list seems outside the scope of this"country_selection" branch

FrnRmn commented on src/main.py:
```diff
@@ -1,21 +1,69 @@
+
+args = cli.get_args()
+
+if args.list:
```
I would extract the next rows as a function outside main.py

FrnRmn replied:
possible?

FrnRmn reviewed (commented):
Looks good in general, thanks

FrnRmn commented on src/energymonitor/exceptions.py:
```diff
@@ -0,0 +1,9 @@
+"""Define custom exceptions"""
```
to be removed

FrnRmn reviewed (changes requested):
I think this is the only necessary change, thanks

MattiaSilvestri commented:
Made some changes:
- removed the exceptions module;
- created the "core" module which will host the scripts handling the core functionalities of the apps. I moved the main there and created a new "cpuCO2" script which will be used only for launching.

FrnRmn commented:
so many changes, basically a new pr haha :D
I might need a call to clarify a few things

MattiaSilvestri commented:
Yeah sorry ahah, I tried to add the things you asked, but I'm still not very sure about the core package, I just didn't know how to better separate the two parts of the main (user arguments handling and main() funciton), but yeah we can discuss it
